lib/core/versions/latest/util/JsObject.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Class containing JavaScript object operations.
+ */
+export default class JsObject {
+  /**
+   * Deep copies the given input.
+   */
+  public static deepCopyObject (input: any): any {
+    if (typeof input !== 'object') {
+      return input;
+    }
+
+    const deepCopy: any = Array.isArray(input) ? [] : {};
+
+    for (const key in input as object) {
+      const value = input[key];
+
+      // Recursively deep copy properties.
+      deepCopy[key] = JsObject.deepCopyObject(value);
+    }
+
+    return deepCopy;
+  }
+
+  /**
+   * Clears all the properties in the given object.
+   */
+  public static clearObject (input: any) {
```

Comment:
we can have a test that makes sure the input turns into an empty object

Reply:
I added a test already to make sure the usage of this method shows that it is working which in my opinion is more meaningful. But sure, I'll add a unit test specifically for this method.